### PR TITLE
add some more service credential docs

### PR DESCRIPTION
**Pull request by minrk**
working toward #4553

@consideRatio what do you think is still missing from #4553?

### CODE REVIEW

File: docs/source/reference/services.md [hunk 1]
```diff
@@ -30,15 +29,16 @@ Currently, these characteristics distinguish two types of Services:
 A Service may have the following properties:
 
 - `name: str` - the name of the service
-- `admin: bool (default - false)` - whether the service should have
-  administrative privileges
-- `url: str (default - None)` - The URL where the service is/should be. If a
-  url is specified for where the Service runs its own web server,
+- `url: str (default - None)` - The URL where the service is/should be.
+  If a url is specified for where the Service runs its own web server,
```
Comment (consideRatio) [inline docs/source/reference/services.md:33]: I don't understand the URL description so well.

> If a url is specified for where the Service runs its own web server

File: docs/source/reference/services.md [hunk 2]
```diff
@@ -30,15 +29,16 @@ Currently, these characteristics distinguish two types of Services:
...
-- `api_token: str (default - None)` - For Externally-Managed Services you need to specify
-  an API token to perform API requests to the Hub
+- `api_token: str (default - None)` - For Externally-Managed Services,
+  you need to specify an API token to perform API requests to the Hub.
+  For Hub-managed services, this token is generated at startup,
+  and available via `$JUPYTERHUB_API_TOKEN`.
+  For OAuth services, this is the client secret.
 - `display: bool (default - True)` - When set to true, display a link to the
-  service's URL under the 'Services' dropdown in user's hub home page.
+  service's URL under the 'Services' dropdown in users' hub home page.
```
Comment (consideRatio) [inline docs/source/reference/services.md:41]: display is related to url, without an url, display is irrelevant I think.

File: docs/source/reference/services.md [hunk 3]
```diff
@@ -30,15 +29,16 @@ Currently, these characteristics distinguish two types of Services:
 A Service may have the following properties:
```
Comment (consideRatio) [inline docs/source/reference/services.md:29]: This list doesn't mention `oauth_client_id` or `oauth_redirect_uri` yet

Comment (minrk) [reply docs/source/reference/services.md:29]: Good point, Added.

File: docs/source/reference/services.md (hunk 2 above)
Comment (minrk) [reply docs/source/reference/services.md:41]: Added that note

File: docs/source/reference/services.md (hunk 1 above)
Comment (minrk) [reply docs/source/reference/services.md:33]: Took another stab at it. Let me know what you think.

File: docs/source/reference/services.md [hunk 4]
```diff
@@ -30,44 +29,54 @@ Currently, these characteristics distinguish two types of Services:
...
   and shouldn't need extra prompts for login.
+- `oauth_client_id: str (default - 'service-$name')` -
+  This never needs to be set,
+  but you can specify a service's OAuth client id.
+- `oauth_redirect_uri: str (default: '/services/$name/oauth_redirect')` -
+  Set the OAuth redirect URI.
+  Required if it differs from the default proxied prefix URI,
+  e.g. the redirect handler is not at `/$prefix/oauth_redirect`,
+  and/or the service is not to be added to the proxy at `/services/$name`
+  (i.e. `url` is not set, but there is still a public web service using OAuth).
```
Comment (consideRatio) [inline docs/source/reference/services.md:59]: - Is `$prefix` the same as `services/$name`? I think we should avoid introducing `$prefix` and just be explicit about that if so.
- `/services/$name` is written here, but its written as `/services/:name` above

File: docs/source/reference/services.md [hunk 5]
```diff
@@ -30,44 +29,54 @@ Currently, these characteristics distinguish two types of Services:
...
   skip the OAuth confirmation page when users access this service.
-
   By default, when users authenticate with a service using JupyterHub,
   they are prompted to confirm that they want to grant that service
   access to their credentials.
   Skipping the confirmation page is useful for admin-managed services that are considered part of the Hub
   and shouldn't need extra prompts for login.
+- `oauth_client_id: str (default - 'service-$name')` -
+  This never needs to be set,
+  but you can specify a service's OAuth client id.
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply docs/source/reference/services.md:53]:
```suggestion
- `oauth_client_id: str (default - 'service-$name')` -
  This never needs to be set, but you can specify a service's OAuth client id.
  It must start with `service-`.
```

File: docs/source/reference/services.md (hunk 4 above)
Comment (consideRatio) [reply docs/source/reference/services.md:59]: I think this section could be omitted as it confused me more than helped me understand.

> proxied prefix URI, e.g. the redirect handler is not at `/$prefix/oauth_redirect`

File: docs/source/reference/services.md [hunk 6]
```diff
@@ -174,6 +183,106 @@ c.JupyterHub.services = [
...
+
+- owned by the authenticated user
+- associated with the oauth client of the service
+- governed by the service's `oauth_client_allowed_scopes` configuration
+
+This token enables the service to act _on behalf of_ the user.
+
+When an oauthenticated service makes a request to the Hub (or other Hub-authenticated service), it has two credentials available to authenticate the request:
+
+- the service's own api*token, which acts \_as* the service,
```
Comment (consideRatio) [inline docs/source/reference/services.md:222]: Formatting issue here I think

**Review by consideRatio — COMMENTED** [review]
I just read through the service credentials examples and I think they are great!!! I've slowly gained more and more understanding about this, and this helped me understad the permission system a bit better once more.

**Comment by consideRatio** [comment]
❤️ 🎉 wieee thank you for working this!!

**Pull request merged**